WordPress/Classes/Services/LoginFacade.m

```diff
@@ -0,0 +1,125 @@
+#import "LoginFacade.h"
+#import "LoginFields.h"
+#import "NSString+Helpers.h"
+#import "NSURL+IDN.h"
+#import "WordPressComOAuthClientFacade.h"
+#import "WordPressXMLRPCApiFacade.h"
+
+
+@interface LoginFacade () {
+    id<LoginFacadeDelegate> _delegate;
+    id<WordPressComOAuthClientFacade> _wordpressComOAuthClientFacade;
+    id<WordPressXMLRPCApiFacade> _wordpressXMLRPCApiFacade;
+}
+
+@end
+
+
+@implementation LoginFacade
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self) {
+        [self initializeServices];
+    }
+    return self;
+}
+
+- (void)initializeServices
+{
+    _wordpressComOAuthClientFacade = [WordPressComOAuthClientFacade new];
+    _wordpressXMLRPCApiFacade = [WordPressXMLRPCApiFacade new];
+}
+
+- (id<LoginFacadeDelegate>)delegate {
```

Comment:
The reason is that this code is called during the initializer and we don't want to be accessing the property directly during initialization.